docs/guides/custom-schema.md

```diff
@@ -0,0 +1,237 @@
+
+> **Hint:** You can [learn more about schemas and types](https://graphql.org/learn/schema/) on <https://graphql.org>. You can also find a reference of all the types generated by Keystone in built-in GraphQL Playground's Docs tab, accessible via the Admin UI.
+
+```graphql
+type IncrementPageViewsOutput {
+  # The new page views after the mutation.
+  currentViews: Int!
+
+  # The time and date at which the mutation was executed.
+  timestamp: String!
+}
+```
+
+We can register this type with Keystone using the `keystone.extendGraphQLSchema()` method, like so. Note the `type` key simply takes the definition as a string:
+
+```javascript title=/lists/Page.js
+keystone.extendGraphQLSchema({
+  types: [
+    {
+      type: 'type IncrementPageViewsOutput { currentViews: Int!, timestamp: String! }',
+    },
+  ],
+});
+```
+
+## The mutation
+
+Now, let's define our custom mutation:
+
+```graphql
+incrementPageViews(id: ID!): IncrementPageViewsOutput
+```
+
+This defines a mutation called `incrementPageViews`that accepts a single mandatory argument called `id` of type `ID` (an internal GraphQL type). This is the id of the specific `Post` whose views you which to increment. Finally, the mutation returns an object of our custom `IncrementPageViewsOutput` type.
+
+Now, to wire it in:
+
+```diff title=/lists/Page.js
+ keystone.extendGraphQLSchema({
+   types: [
+     {
+       type: 'type IncrementPageViewsOutput { currentViews: Int!, timestamp: String! }',
+     },
+   ],
++  mutations: [
++    {
++      schema: 'incrementPageViews(id: ID!): IncrementPageViewsOutput',
++      resolver: handleIncrementPageViews,
++    },
++  ],
+ });
+```
+
+Custom mutations (and queries) take a `schema` and a `resolver` key. Like with the custom type, `schema` is simply the custom mutation definition, while `resolver` is a function which tells Keystone how to execute this mutation. In our case, it must return a `IncrementPageViewsOutput` object.
+
+> **Note:** Refer to the [`keystone.extendGraphQLSchema`](/packages/keystone/README.md#extendgraphqlschemaconfig) API documentation for details on the `resolver` function's signature.
+
+Let's define our resolver function:
+
+```javascript title=/lists/Page.js
+const incrementPageViews = async (_, { id }) => {
+  const list = keystone.lists.Page;
+
+  const oldItem = await list.adapter.findById(id);
+  const newItem = await list.adapter.update(id, {
+    ...oldItem,
+    views: (oldItem.views || 0) + 1,
+  });
+
+  return {
+    currentViews: newItem.views,
+    timestamp: new Date().toISOString(),
+  };
+};
+```
+
+> **Note:** The value of `views` may be `undefined` initially, so before we increment it we make sure to convert any `falsey` values to `0`.
+
+Our custom mutation is now available to the client! It can be utilized like so:
+
+```graphql
+mutation incrementPageViews($id: ID!) {
+  incrementPageViews(id: $id) {
+    currentViews
+    timestamp
+  }
+}
+```
+
+### A note on access control
```

Comment:
Honestly, I think this section should probably be dropped. It's kinda tangential. Thoughts?